added bench_importasdm.py

bgarwood opened the pull request.

bgarwood committed:
added bench_importasdm.py

amcnicho reviewed (commented):
These new tests on my sandbox take about 40m to complete all iterations.
```
real    41m9.271s
user    23m42.139s
sys     3m29.037s
```
samples = 6 for every new test in the results.json file generated locally, so it might be worth considering a tweak to the iteration control parameters to bring the total runtime down. 40m for 6 iterations of 25 new tests is also not that much, so it could be fine the way it is -- up to you @smcastro.

Just one of the new tests failed for me (`bench_importasdm.AutocorrASDM.time_autocorr`) with a couple of errors in the traceback that seem to suggest either an improperly configured data repository on my end or a variable referenced before assignment (indeed, possibly both):
```
NameError: name 'outfile' is not defined
...
FileNotFoundError: [Errno 2] No such file or directory: 'AutocorrASDM.ms'
```
To me that seems like it could be a problem with the setup/teardown for that test class.

I think there should also be a version string for each new test, so that minor changes to the code which do not materially change the test function will be considered to have the same result as previously saved test runs (and it's more convenient to reference than the hash of the benchmark function). That should be quick to add; we've been using the issue name and that seems to work well so far.

Apart from that the new test script looks good overall. Thanks @bgarwood!

bgarwood commented:
@amcnicho There's a bug in that script (after I ripped out some irrelevant parts from the associated test) so that "output" is used without being set. That cause importasm to fail before it creates the output ms which then leads to the second error in teardown when it tries to remove the expected output ms that doesn't exist.

Can you point me at an existing bench* script that uses a version string as you described? I could then also edit bench_importasdm.py to do something similar. Thanks.

bgarwood committed:
outfile argument in time_autocorr was not being set correctly

bgarwood commented:
I believe the last commit to this branch fixes the issues with time_autocorr

amcnicho commented:
That test is still failing in my sandbox even after ffc31cb9b3c53b51aa442d744bc3bb5cebd5cf99, and it looks like a similar issue; no `FileNotFoundError` in the traceback this time though.

Any of the newer test files (e.g., [benchmarks/bench_fringefit.py](https://github.com/casangi/casabench/blob/main/benchmarks/bench_fringefit.py#L41)) and the [asv docs](https://asv.readthedocs.io/en/stable/writing_benchmarks.html#benchmark-versioning) show the use of version strings to override the default "hash of the test code" version setting.

bgarwood commented:
Can you show the traceback this time? It works for me and the lack of any FileNotFoundError suggests that importasdm actually ran that time (the output ms was created). I wonder why it works for me? 

Also, is there a way to do a specific time function in a specific bench script (as opposed to the entire bench script, which I know how to do).

amcnicho commented:
Yes, you can just keep adding specificity to the `--bench` argument and that will drill down on an individual test case. Here is how I ran the single test (after confguring the asv.conf.json to match the 6.6.0.16 tag of course):
`asv run 6.6.0.16^! --machine "NRAO workstation" --bench bench_importasdm.AutocorrASDM.time_autocorr --quick --show-stderr --verbose`
and here's the traceback:
```
[ 33.33%] ··· bench_importasdm.AutocorrASDM.time_autocorr                                                                   failed
[ 33.33%] ···· Traceback (most recent call last):
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 1293, in main_run_server
                   main_run(run_args)
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 1171, in main_run
                   benchmark.do_teardown()
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 566, in do_teardown
                   teardown(*self._current_params)
                 File "/export/home/mano/Workspace/clones/casabench/benchmarks/bench_importasdm.py", line 146, in teardown
                   os.remove(outfile)
               NameError: name 'outfile' is not defined
               asv: benchmark failed (exit status 1)
```
I thought the difference could be because you are running without the `--quick` option, but removing that doesn't work for me either (I get the same failure, plus that original MS not found traceback):
```
               During handling of the above exception, another exception occurred:
               
               Traceback (most recent call last):
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 1293, in main_run_server
                   main_run(run_args)
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 1171, in main_run
                   benchmark.do_teardown()
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/site-packages/asv/benchmark.py", line 566, in do_teardown
                   teardown(*self._current_params)
                 File "/export/home/mano/Workspace/clones/casabench/benchmarks/bench_importasdm.py", line 141, in teardown
                   shutil.rmtree(self.asdm_name)
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/shutil.py", line 709, in rmtree
                   onerror(os.lstat, path, sys.exc_info())
                 File "/users/amcnicho/miniconda3/envs/asv/lib/python3.8/shutil.py", line 707, in rmtree
                   orig_st = os.lstat(path)
               FileNotFoundError: [Errno 2] No such file or directory: 'AutocorrASDM'
               asv: benchmark failed (exit status 1)
```

bgarwood committed:
fix bug in AutocorrASDM teardown, added versions, tweaked ASV parameters

bgarwood commented:
I was unaware of "--show-stderr" and that explains why I didn't see the errors .

I've committed a change which I believe fixes the errors you've been seeing. I added the version tags to each time function.
I also changed rounds to 1, and min_round_count to 1 from 3 and 5 respectively.

I'm pretty sure min_round_count of 5 was too high but the other 2 values seem largely random to me. None of them have any magic quality and it would be fine if they were set to whatever seems appropriate to someone else.

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co',with_pointing_correction=True)
+
+    def time_evla_apply2(self):
```
This one could be renamed to time_apply_flags()

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co', with_pointing_correction=True,
+                   process_flags=True, applyflags=True, savecmds=False, flagbackup=False)
+
+    def time_evla_apply3(self):
```
suggest renaming to something such as time_scans or time_scan_selection. It is not applying the flags on the fly therefore the word "apply" should not appear.

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2', ocorr_mode='co', with_pointing_correction=True,
+                   process_flags=True, applyflags=True, savecmds=True, outfile=self.cmdfile, flagbackup=False)
+                
+    def time_evla_apply3_flagdata(self):
```
rename

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+        if os.path.exists(self.outfile):
+            os.remove(outfile)
+    
+    def time_autocorr(self):
```
suggest renaming: time_autocorr_outfile

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='3', savecmds=True, outfile=self.outfile)
+
+    def time_flagautocorr1(self):
```
rename

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='3', applyflags=True)
+
+    def time_flagautocorr3(self):
```
rename

smcastro commented on benchmarks/bench_importasdm.py:
```diff
@@ -0,0 +1,243 @@
+
+        importasdm(asdm=self.asdm_name, vis=self.ms_name, lazy=True, scans='0:1~3')
+
+    def time_aca(self):
```
rename to time_scans

smcastro reviewed (changes requested):
Bob, the content of the tests is good and has good coverage compared to the original tests. I only would like you to please rename the class M51 to Basic and for some test cases, please find more descriptive names that can already tell what is being run when looked at the asc test menu. Use parameters or functionality to describe the test names. I give some examples below. Thank you

bgarwood commented:
Thanks. My older daughter is having surgery today with an overnight stay in the hospital. I'll probably be able to take care of this on Friday.

Sep 27, 2023 8:48:27 AM smcastro ***@***.***>:

> 
> ***@***.**** requested changes on this pull request.
> 
> Bob, the content of the tests is good and has good coverage compared to the original tests. I only would like you to please rename the class M51 to Basic and for some test cases, please find more descriptive names that can already tell what is being run when looked at the asc test menu. Use parameters or functionality to describe the test names. I give some examples below. Thank you
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338537082]:
> 
>> +
> +from casatools import ctsys
> +from casatasks import importasdm, casalog
> +
> +# ASV iteration control (https://asv.readthedocs.io/en/stable/benchmarks.html#benchmark-attributes)
> +number = 1            # i.e., always run the setup and teardown methods
> +repeat = (1, 2, 30.0) # between 1 and 2 iterations per round w/ soft cutoff (start no new repeats) past 1m
> +rounds = 3            # amount of instances a "repeat block" is run to collect samples
> +min_run_count = 5     # enforce the min_repeat * rounds setting is met
> +timeout = 3600        # conservative 1hr hard cap for duration of a single test execution
> +
> +datapath = ctsys.resolve('unittest/importasdm/')
> +
> +# benchmarks grouped by ASDM used
> +
> +class M51():
> 
> I think it is better a more descriptive name for this class. I suggest class Basic() and add the test cases that use the M51 dataset with basic parameters such as default, etc.
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338540426]:
> 
>> +    asdm_name = 'X_osro_013.55979.93803716435'
> +    ms_name = asdm_name + '.ms'
> +    cmdfile = ms_name.replace('.ms','_cmd.txt')
> +
> +    def setup(self):
> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        if (os.path.exists(self.ms_name+'.flagversions')):
> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if (os.path.exists(self.cmdfile)):
> +            os.remove(self.cmdfile)        
> +    
> +    def time_evlatest1(self):
> 
> It is better to use more descriptive test case names without an index. This one could be called
> time_evla_pc_true() or time_pointing_correction_true() without the 'evla' because it is already the name of the class. It is up to you to keep or not the name evla.
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338541444]:
> 
>> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        if (os.path.exists(self.ms_name+'.flagversions')):
> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if (os.path.exists(self.cmdfile)):
> +            os.remove(self.cmdfile)        
> +    
> +    def time_evlatest1(self):
> +        '''test_evlatest1 - test of importing evla data, with_pointint_correction=True is normal for EVLA data'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co',with_pointing_correction=True)
> +
> +    def time_evla_apply2(self):
> 
> This one could be renamed to time_apply_flags()
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338544301]:
> 
>> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if (os.path.exists(self.cmdfile)):
> +            os.remove(self.cmdfile)        
> +    
> +    def time_evlatest1(self):
> +        '''test_evlatest1 - test of importing evla data, with_pointint_correction=True is normal for EVLA data'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co',with_pointing_correction=True)
> +
> +    def time_evla_apply2(self):
> +        '''test_evla_apply2 - test of importing evla data and applying the online flags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=True, applyflags=True, savecmds=False, flagbackup=False)
> +
> +    def time_evla_apply3(self):
> 
> suggest renaming to something such as time_scans or time_scan_selection. It is not applying the flags on the fly therefore the word "apply" should not appear.
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338545116]:
> 
>> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co',with_pointing_correction=True)
> +
> +    def time_evla_apply2(self):
> +        '''test_evla_apply2 - test of importing evla data and applying the online flags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2',ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=True, applyflags=True, savecmds=False, flagbackup=False)
> +
> +    def time_evla_apply3(self):
> +        '''test_evla_apply3 - test of importing evla data, two difference scans, do not apply online flags'''
> +        
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2,13', ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=False,flagbackup=False)
> +
> +    def time_evla_apply5(self):
> 
> suggestion to rename to time_all_scans
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338546933]:
> 
>> +                   process_flags=False,flagbackup=False)
> +
> +    def time_evla_apply5(self):
> +        '''test_evla_apply5 - test of importing evla data: all scans, do not process flags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=False, flagbackup=False)
> +
> +    def time_evla_savepars(self):
> +        '''test_evla_savepars - test importing evla data: save the flag commands and do not apply'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name,scans='11~13', ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=True, applyflags=False, savecmds=True, flagbackup=False)
> +
> +
> +    def time_evla_apply1_flagdata(self):
> 
> Please find a name without an index and use descriptive names coming from parameters for example.
> time_scan_apply_save_flags()
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338547240]:
> 
>> +                   process_flags=False, flagbackup=False)
> +
> +    def time_evla_savepars(self):
> +        '''test_evla_savepars - test importing evla data: save the flag commands and do not apply'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name,scans='11~13', ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=True, applyflags=False, savecmds=True, flagbackup=False)
> +
> +
> +    def time_evla_apply1_flagdata(self):
> +        '''test_evla_apply1_flagdata - test of importing evla data, apply onlineflags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='2', ocorr_mode='co', with_pointing_correction=True,
> +                   process_flags=True, applyflags=True, savecmds=True, outfile=self.cmdfile, flagbackup=False)
> +                
> +    def time_evla_apply3_flagdata(self):
> 
> rename
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338547748]:
> 
>> +
> +        
> +class Evla_ephemeris():
> +    # EVLA sdm with ephemeris
> +    asdm_name = 'polyuranus'
> +    ms_name = asdm_name + '.ms'
> +
> +    def setup(self):
> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        shutil.rmtree(self.ms_name+'.flagversions')
> +    
> +    def time_evlatest2(self):
> 
> rename
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338548500]:
> 
>> +    asdm_name = 'AutocorrASDM'
> +    ms_name = asdm_name + '.ms'
> +    outfile = 'scanflags.txt'
> +
> +    def setup(self):
> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        if (os.path.exists(self.ms_name+'.flagversions')):
> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if os.path.exists(self.outfile):
> +            os.remove(outfile)
> +    
> +    def time_autocorr(self):
> 
> suggest renaming: time_autocorr_outfile
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338548715]:
> 
>> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        if (os.path.exists(self.ms_name+'.flagversions')):
> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if os.path.exists(self.outfile):
> +            os.remove(outfile)
> +    
> +    def time_autocorr(self):
> +        '''test_autocorr - importasdm: auto-correlations should be written to online flags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='3', savecmds=True, outfile=self.outfile)
> +
> +    def time_flagautocorr1(self):
> 
> rename
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338548852]:
> 
>> +        if (os.path.exists(self.ms_name+'.flagversions')):
> +            shutil.rmtree(self.ms_name+'.flagversions')
> +        if os.path.exists(self.outfile):
> +            os.remove(outfile)
> +    
> +    def time_autocorr(self):
> +        '''test_autocorr - importasdm: auto-correlations should be written to online flags'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='3', savecmds=True, outfile=self.outfile)
> +
> +    def time_flagautocorr1(self):
> +        '''test_flagautocorr1 - importasdm: test that auto-correlations from online flags are correctly flagged'''        
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, scans='3', applyflags=True)
> +
> +    def time_flagautocorr3(self):
> 
> rename
> 
> ----------------------------------------
> 
> In benchmarks/bench_importasdm.py[https://github.com/casangi/casabench/pull/20#discussion_r1338549507]:
> 
>> +    ms_name = asdm_name + '.ms'
> +
> +    def setup(self):
> +        shutil.copytree(os.path.join(datapath,self.asdm_name), self.asdm_name)
> +
> +    def teardown(self):
> +        shutil.rmtree(self.asdm_name)
> +        shutil.rmtree(self.ms_name)
> +        shutil.rmtree(self.ms_name+'.flagversions')
> +               
> +    def time_aca_lazy(self):
> +        '''test6_lazy1 - lazy fill: Test good ACA ASDM with mixed pol/channelization input with default filler in lazy mode'''
> +
> +        importasdm(asdm=self.asdm_name, vis=self.ms_name, lazy=True, scans='0:1~3')
> +
> +    def time_aca(self):
> 
> rename to time_scans
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/casangi/casabench/pull/20#pullrequestreview-1646359459], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AABL7MR4ZUZOVARYV7NDJJ3X4QN6PANCNFSM6AAAAAA46KTP3A].
> You are receiving this because you were mentioned.
> 
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/casangi/casabench/pull/20#pullrequestreview-1646359459", "url": "https://github.com/casangi/casabench/pull/20#pullrequestreview-1646359459", "name": "View Pull Request" }, "description": "View this Pull Request on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

bgarwood committed:
renamed functions and one class, removed one function

bgarwood commented:
I've renamed multiple functions following @smcastro comments. I also removed one time function as it duplicated an importasdm call covered by another function (they came from different test functions in test script and I missed that the importasdm calls were identical - the testing there involved the downstream flagging).

amcnicho commented:
All the new/renamed tests pass now, here are the updated timings when run in my sandbox:
```
· Running 24 total benchmarks (1 commits * 1 environments * 24 benchmarks)
[  0.00%] · For casa6 commit bd757754 <6.6.3.4>:                                                                                  
[  0.00%] ·· Benchmarking virtualenv-py3.8-pip+casatasks6.6.3.4-pip+casatestutils6.6.3.4                                          
[  0.00%] ··· Importing benchmark suite produced output:                                                                          
[  4.17%] ··· bench_importasdm.Aca.time_default           6.38±0.2s
[  8.33%] ··· bench_importasdm.Aca.time_default_lazy           6.81±0.7s
[ 12.50%] ··· bench_importasdm.Alma_12m.time_auto_only           8.06±0s
[ 16.67%] ··· bench_importasdm.Alma_12m.time_auto_only_lazy           8.63±0.06s
[ 20.83%] ··· bench_importasdm.Alma_12m.time_bdfflags_true           23.8±0s
[ 25.00%] ··· bench_importasdm.Alma_12m.time_bdfflags_true_lazy           21.4±0s
[ 29.17%] ··· bench_importasdm.Alma_12m.time_default           14.7±0s
[ 33.33%] ··· bench_importasdm.Alma_12m.time_default_lazy           11.7±0.05s
[ 37.50%] ··· bench_importasdm.AutocorrASDM.time_applyflags_true           4.25±0.1s
[ 41.67%] ··· bench_importasdm.AutocorrASDM.time_process_flags_false           2.96±0.02s
[ 45.83%] ··· bench_importasdm.AutocorrASDM.time_savecmds_true           3.59±0.02s
[ 50.00%] ··· bench_importasdm.Basic.time_all_defaults           4.59±0.09s
[ 54.17%] ··· bench_importasdm.Basic.time_lazy_pc_true           .60±0.02s
[ 58.33%] ··· bench_importasdm.Basic.time_pc_true           4.40±0.03s
[ 62.50%] ··· bench_importasdm.Evla.time_all_scans           4.42±0s
[ 66.67%] ··· bench_importasdm.Evla.time_applyflags_true           4.41±0.07s
[ 70.83%] ··· bench_importasdm.Evla.time_pc_true           3.15±0.01s
[ 75.00%] ··· bench_importasdm.Evla.time_process_applyflags_savecmds_true           4.22±0s
[ 79.17%] ··· bench_importasdm.Evla.time_process_flags_savecmds_true           3.23±0.01s
[ 83.33%] ··· bench_importasdm.Evla.time_process_flags_true_applyflags_false   3.24±0.01s
[ 87.50%] ··· bench_importasdm.Evla.time_scan_selection           3.06±0.02s
[ 91.67%] ··· bench_importasdm.Evla_ephemeris.time_polynomial_ephem           1.31±0m
[ 95.83%] ··· bench_importasdm.Singledish.time_auto_only           6.82±0.04s
[100.00%] ··· bench_importasdm.Singledish.time_auto_only_lazy           6.53±0.02s

real    12m34.151s
user    7m38.415s
sys     0m54.920s
```
LGTM, thank you for working on this new script @bgarwood.
